lib/ansible/modules/packaging/language/pip.py

```diff
@@ -486,23 +486,21 @@ def __init__(self, name_string, version_string=None):
         self._plain_package = False
         self.package_name = name_string
         self._requirement = None
-        if name_string.startswith('file:') or _is_vcs_url(name_string):
-            return
 
-        self._plain_package = True
         if version_string:
             version_string = version_string.lstrip()
             separator = '==' if version_string[0].isdigit() else ' '
             name_string = separator.join((name_string, version_string))
         try:
             self._requirement = Requirement.parse(name_string)
+            # old pkg_resource will replace 'setuptools' with 'distribute' when it already installed
+            if self._requirement.project_name == "distribute":
+                self.package_name = "setuptools"
+            else:
+                self.package_name = self._requirement.project_name
+            self._plain_package = True
         except ValueError as e:
-            raise ValueError("Can not parse package name '%s', error: '%s'" % (name_string, to_native(e)))
```

Comment:
What is the reason for removing error handling?

Reply:
Like @robinro said, it's not necessary, `pip` can handle the invalid package name.
Another thing is, users can input a file name or a directory name to `name`, like`file:\\xxxx`, 
 `django.tar.gz` or `./django/`, we should only parse the plain package name and leave all other types of name to `pip`, let `pip` decide whether a file or directory is valid to install.

Reply:
ok